cpp/src/io/parquet/page_hdr.cu

```diff
@@ -148,10 +148,20 @@ __device__ void skip_struct_field(byte_stream_s* bs, int field_type)
  * @return `kernel_mask_bits` value for the given page
  */
 __device__ decode_kernel_mask kernel_mask_for_page(PageInfo const& page,
-                                                   ColumnChunkDesc const& chunk)
+                                                   ColumnChunkDesc const& chunk,
+                                                   int use_fixed_op)
 {
   if (page.flags & PAGEINFO_FLAGS_DICTIONARY) { return decode_kernel_mask::NONE; }
-
+  // TODO: once we remove `use_fixed_op` we should also simplify this
+  // to use some functions (e.g. is_fixed_width, is_dictionary)
+  if (use_fixed_op != 0 && !is_string_col(chunk) && chunk.max_nesting_depth == 1 &&
```

Comment:
this is the kind of boolean that @vuule would make me break up into bite size chunks :)

Reply:
so it's true, hurt people hurt people :cry: 
https://github.com/rapidsai/cudf/pull/15081#discussion_r1503443314

Reply:
`USE_FIXED_OP` and this boolean no longer exists: https://github.com/rapidsai/cudf/pull/15159/commits/8e9a98105cbe83b19989977a42c5be39036f30bc